app/views.py

```diff
@@ -0,0 +1,207 @@
+#! /usr/bin/env python
+from flask_restful import abort, Resource, reqparse, marshal_with
+from flask import abort, jsonify, request
+from . import db, expiry_time
+from app.models import User, Bucketlist, Item
+from app.authenticate import multi_auth, g
+from app.validate_format import (itemformat, bucketlistformat, validate_string,
+                                 save, delete, is_not_empty)
+
+
+class LoginUser(Resource):
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('username', type=str, required=True,
+                                   location='json', help="username required")
+        self.reqparse.add_argument('password', type=str, required=True,
+                                   location='json', help="password required")
+        super(LoginUser, self).__init__()
+
+    def post(self):
+        args = self.reqparse.parse_args()
+        username = args.username
+        password = args.password
+        user = User.query.filter_by(username=username).first()
+        if not user or not user.verify_password(password):
+            return {"message": "wrong login details"}, 401
+        token = user.generate_confirmation_token(expiry_time)
+        return {"token": token.decode("ascii")}, 200
+
+
+class RegisterUser(Resource):
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument("username", type=str, required=True,
+                                   location="json", help="username required")
+        self.reqparse.add_argument("password", type=str, required=True,
+                                   location="json", help="password required")
+        self.reqparse.add_argument("email", type=str, required=True,
+                                   location="json", help="email required")
+        super(RegisterUser, self).__init__()
+
+    def post(self):
+        args = self.reqparse.parse_args()
+        username = args["username"]
+        password = args["password"]
+        email = args["email"]
+        if not validate_string(username):
+            return {"message": ("only numbers, letters, '-','_' allowed"
+                                " in username")}, 400
+        if not is_not_empty(username, password, email):
+            return {"message": "no blank fields allowed"}, 400
+        # validate the email field
+        if not ("@" in email):
+            return {"message": "email is invalid"}, 400
```

Comment:
that is not a good way of validating your email. Check out this site to help you write proper Regex expressions https://regex101.com/